replica/table.cc

```diff
@@ -439,6 +439,14 @@ sstables::shared_sstable table::make_sstable() {
     return make_sstable(sstables::sstable_state::normal);
 }
 
+db_clock::time_point table::get_truncation_time() const {
+    if (!_truncated_at) [[unlikely]] {
+        on_internal_error(dblog, ::format("truncation time is not set, table {}.{}",
```

Comment:
but what about `distributed_loader::init_system_keyspace`? some system tables also can be truncated, and we can't load truncation records until we load the system table 'truncated'. We could have two calls to `load_truncation_times` inside `parse_system_tables` and `distributed_loader::init_system_keyspace`, but it's wasteful and doesn't seem like much of an improvement.

Reply:
We'd have to cover init_system_keyspace() too. As soon as we call sys_ks.make(), we can read from system.truncated. There is a common path shared by both system and non-system: distributed_loader::populate_keyspace(). We could call get_truncation_record() for each table. Commitlog replayer already does it, so it doesn't add much extra cost. The replayer queries will hit in cache. Or we could change the replayer to rely on the fact that the record is cached in the table object.

Reply:
> It would be better to structure the code so that this cannot happen, by loading truncation records when loading tables (database::parse_system_tables()). 

I still don't see a way to achieve that - we can't initialise `_truncated_at` in the table constructor since 'truncate' table might not have been loaded yet.

Reply:
We don't have to initialize it in the constructor. Just like we don't attach sstables in the constructor.